### ISSUE TITLE
1D constructor

### ISSUE THREAD

**Issue by nshervt** [kymatio/kymatio#377]
I was trying to compare 1D filter constructor in scatnet with kymatio, and I couldn't make a one-to-one correspondence between the two. Here are a handful of items that I'm confused about:
As an example, I tried to regenerate the following filter in kymatio:
```
filter_size = 16
option.boundary = 'per';
option.J = 3;
option.P = 0;
filter = morlet_filter_bank_1d(filter_size, option);
```
the equivalent function in kymatio would be
```
scattering_filter_factory(J_support, J_scattering, Q, r_psi=math.sqrt(0.5),
                              criterion_amplitude=1e-3, normalize='l1',
                              to_torch=False, max_subsampling=None,
                              sigma0=0.1, alpha=5., P_max=5, eps=1e-7)
```
I understand that $filter_size = 2^{J_support}$, so it should be 4, then the default value for Q in scatnet is 1. I also noticed that the default value for sigma0 is $2/\sqrt(3)$ in scatnet while it is $0.1$ is kymatio. I also think J_scattering should be 3. 

Now with my assumptions, I check the outputs of these two functions: for scatnet's morlet_filter_bank_1d I have:

filter.phi.filter:    coefft: {[16×1 double]  [8×1 double]  [4×1 double]  [2×1 double]}
filter.psi.filter:  cell array :  {1×1 struct}    {1×1 struct}    {1×1 struct} , where each struct has same form as filter.phi.filter.

let's right down first vector in phi:
[-0.0000,    0.0715,    0.2331,    0.5101,    0.8683,    1.1880,    1.3186,    1.1903,    0.8745,    0.5228,    0.2539,    0.0985,    0.0263,   -0.0041,   -0.0180,   -0.0206]

for kymatio's scattering_filter_factory, I get 
phi  :      {[16×1 double]  [8×1 double]}
psi1 :      {[16×1 double]  [16×1 double]  [16×1 double]  [16×1 double]}
psi2 :      {[16×1 double]  [16×1 double]  [16×1 double]  [16×1 double]}

again, I check the first element in phi:
[1.00000000e+000 3.72665317e-006 1.92874985e-022 1.38634329e-049
 1.38389653e-087 1.91855567e-136 3.69388307e-196 9.87710872e-267
 0.00000000e+000 9.87710872e-267 3.69388307e-196 1.91855567e-136
 1.38389653e-087 1.38634329e-049 1.92874985e-022 3.72665317e-006]

which is essentially a zero vector. So how can I regenerate a filter similar to the one defined in scatnet on top using kymatio. Lastly, why 1D low pass filters in kymatio have 2 orders? For instance in 2D scattering, we only have the same variable for low pass filtering and we decide based on 'j' what to apply. Sorry for lengthy question and thanks!

**Comment by edouardoyallon** [comment]
@lostanlen @janden this is surprising, can you coment on this? I'm not familiar with 1D-kymatio, are those filters in the Fourier domain?

**Comment by nshervt** [comment]
@edouardoyallon Yes, according to comments in scattering_filter_factory function:

> Builds in Fourier the Morlet filters used for the scattering transform.

**Comment by AndreuxMath** [comment]
Hi @nshervt ,

Thank you for your feedback!

> again, I check the first element in phi: (...) which is essentially a zero vector.

It looks like the filter generation in kymatio is not working well for very small input sizes. I think it was only tested for large `J_support`, something like `J_support = 13`. This explains the zero outputs. I can help on fixing this one.

> So how can I regenerate a filter similar to the one defined in scatnet on top using kymatio.

As you noticed there's a tiny difference in the `sigma0` value, just as there is a tiny difference in the formula used to generate the gaussians ( [`gauss_1d`](https://github.com/kymatio/kymatio/blob/d81e6cc5e978aa062bd7d1ce84d04542e0d3bc16/kymatio/scattering1d/filter_bank.py#L169) and [`morlet1d`](https://github.com/kymatio/kymatio/blob/d81e6cc5e978aa062bd7d1ce84d04542e0d3bc16/kymatio/scattering1d/filter_bank.py#L75) in kymatio and [`gabor`](https://github.com/scatnet/scatnet/blob/59d935afa20359845282a3518134e24244862c1f/filters/morlet_filter_bank_1d.m#L162) in scatnet ).

Making the aforementioned functions in kymatio closer to scatnet's will be a first step towards having the same filters. However, I am not familiar with the detailed structure of the filter generation in scatnet - at high level they perform roughly the same operations. I suspect there are other non-trivial differences elsewhere (notably for the subsampling).

Fixing this part is likely to be longer and will potentially require rewriting large portions of `kymatio.scattering1d.filter_bank`. @janden or @lostanlen 's help would be needed here.

>  Lastly, why 1D low pass filters in kymatio have 2 orders? 

In kymatio, the second-order scattering uses only wavelets with `Q=1`, while the filters of the first-order transform use `Q >=1`. If `Q` is set to 1, then the second-order filters are redundant with the first-order ones but in general this is not the case. As far as I remember scatnet uses filters with `Q` wavelets per octave for both the first and second-order.

As a consequence, even if the filters of kymatio reproduce those of scatnet, their second-order coefficients will not match for `Q>1`.

**Comment by nshervt** [comment]
Thanks @AndreuxMath for detailed response. 

> This explains the zero outputs. I can help on fixing this one.

I appreciate your help.
